### DISCUSSION TITLE
Polygon editing

### DISCUSSION THREAD

**Discussion by kueda** [davenquinn/Mapboard-GIS#11] [Q&A] [answered]
As far as I can tell, polygons cannot be edited. When I try to use the reshape tool on a polygon I got an error message:

![B6B54E1F-E951-4E81-8BCB-AC5C1BA0B0C1](https://user-images.githubusercontent.com/22691/212583486-84b73d3d-db36-4ede-96b9-9f3dc5038529.png)

And there doesn’t seem to be any way to merge polygons, which seems like a requirement when trying to digitize a large, complex polygon. Is this a bug, or missing functionality, or am I just missing something in the UI?

For what it’s worth, this is a spatialite project in which I’m trying to digitize existing polygons in an MBtiles raster layer.

**Comment by davenquinn** [comment, marked as answer]
Hi @kueda, thanks for the question. There are a few things going on here:

Mapboard GIS is intended to work a bit differently than you are trying to use it right now. Here's the intended mapping workflow:

- Draw lines to enclose polygonal areas. Lines have snapping and reshaping and other niceties that aren't accessible for polygons.
- Use polygons as "stations" to identify which unit is present in a given area
- In the **Topology** panel (accessible from Polygons or settings), the topology can be solved to expand appropriate unit IDs from polygons (largest one in an enclosed area wins) to a enclosed map area.
- As you change the linework, you can update the topology without having to manually move polygons around.

The UI and documentation is not yet well laid out to make this workflow obvious. Any suggestions are welcome.

Because they are mostly intended to be used as "stations", polygon tools are pretty limited relative to the equivalent line tools.

- Polygon reshaping doesn't work right now; the near-term bug is that the tool is still enabled in the UI. See [the *Map Interface* user guide page](https://mapboard-gis.app/docs/map-interface). Drawing polygons without "Allow overlapping polygons" has similar effect. Hopefully, I can enable polygon reshaping soon.
- A "merge polygons" tool is a good idea and will be added to a future release.

**Comment by kueda** [comment]
So you draw your lines and then draw a little polygon inside the area circumscribed by the lines to indicate that that area has a type? If that’s the standard workflow, why not use a point to indicate a face instead of a polygon? As a user who basically just wants to end up with a bunch of non-overlapping polygons, it’s confusing that the “polygon” tool doesn’t actually create those polygons. Maybe a video tutorial would help?

Unrelatedly I’m now getting a "configuration error” whenever I try to view tiles in my mbtiles overlay, not sure if you want a bug report for that or if there’s a way to get more informative log data than the event log shows.

**Reply by davenquinn** [reply]
Yes, that's the idea. The reason to use polygons is essentially that it is easier to draw polygons than points using a pencil. Polygons can also nicely represent the areal extent with a certain unit ID, in some cases. Point support is planned for the future; I also agree a tutorial would help make the intended workflow less ambiguous. I'm also hoping to redesign the UI and rename things to make the intended workflow a bit more obvious.

**Reply by davenquinn** [reply]
For the configuration error, is this a MBTiles dataset you've viewed before? Perhaps there's a bug; I will look into it.

**Reply by kueda** [reply]
Yeah, it's an mbtiles that I generated yesterday and it was working then, but for some reason today it's not. When I make a new project and add that same mbtiles as an overlay I get the same error. The file is 63 MB so I can't attach it here, but I can post it elsewhere if that would help.

**Comment by davenquinn** [comment]
Don’t think there’s a need for you to post it just yet. This is probably a simple bug with layer config values. I will do some testing here with my own MBTiles.
